cli/pkg/cli/cli.go

```diff
@@ -127,6 +128,34 @@ func (c *CLI) ExportResults(ctx context.Context, res *results.Results, errs fami
 	return result
 }
 
+func (c *CLI) WatchForAbort(ctx context.Context, cancel context.CancelFunc, interval time.Duration) {
+	go func() {
+		timer := time.NewTimer(interval)
+		defer timer.Stop()
+
+		for {
+			select {
+			case <-timer.C:
+				aborted, err := c.IsAborted(ctx)
+				if err != nil {
+					log.Errorf("failed to retrieve scan result state: %v", err)
+				}
+				if aborted {
+					cancel()
+					return
+				}
+				timer.Reset(interval)
+			case <-ctx.Done():
+				if !timer.Stop() {
+					<-timer.C
+				}
```

Comment:
can you explain why there is a need to be blocked on timer here?

Reply:
@FrimIdan https://pkg.go.dev/time#Timer.Stop